exec/java-exec/src/test/java/org/apache/drill/TestUnionAll.java

```diff
@@ -899,4 +900,35 @@ public void testUnionAllInWith() throws Exception {
         .build()
         .run();
   }
+
+  @Test // see DRILL-4147
+  public void testUnionAllDistributedMode() throws Exception {
```

Comment:
Similar to this test, could you make the slice_target = 1 for all the other Union-All tests ?  That will give additional coverage.